### ISSUE TITLE
BatchClientImpl okHttp Interceptor Queue Limit at 5

### ISSUE THREAD

**Issue by tslilyai** [Microsoft/EmbeddedSocial-Java-API-Library#18]
The okHttp interceptor used in BatchClientImpl does not seem to be able to intercept more than 5 calls. This causes a batch client that is waiting on more than 5 requests to be added to block forever.

These calls appear not to invoke the intercept() function of the interceptor. Furthermore, the `ServiceCall` returned by an invoked request method, e.g. `createTopicAsync()`, is not cancelled; a response is not returned; and the registered callback is not invoked. This implies that the request was intercepted (or else it would have just gone to the server) but not processed.

Failing example:
```java
EmbeddedSocialBatchedClientImpl batchOps = new EmbeddedSocialBatchedClientImpl.Builder()
                .batchSize(7)
                .baseUrl(ESClient.baseURL)
                .build();
        
for (int i = 0; i < 7; i++) {
      // the interceptor gets invoked here. all 7 calls "succeed" but only 5 invoke the "intercept()" function
      batchOps.getEsClient().getBuildsOperations().getBuildsCurrentAsync(callback);
}
// after this point, the latch() in batchOps still has value 2! issueBatch thus blocks forever
try {
     // the batchOps thread blocks here
     batchOps.issueBatch();
} catch (InterruptedException | IOException e) {
     System.out.println(e.getMessage());
}
```

**Comment by ssaroiu** [comment]
I was able to reproduce this issue. I'm currently working on a fix.

**Comment by ssaroiu** [comment]
Fixed, PR issues, and PR merged (PR #15). Thanks!
